hypnotoad/gui/gui.py

```diff
@@ -152,10 +164,10 @@ def save_options(self):
 
         self.options_file_line_edit.setText(self.filename)
 
-        non_null_options = {k: v for k, v in self.options.items() if v is not None}
-
+        # Could consider (optionally?) saving just non-default options, but safer to save
+        # all options for reproducibility in case defaults are changed in future
```

Comment:
I think I had some issues with saving/loading the yaml when some options were null. I agree that saving everything is better for reproduciblity, but if a value is `None` in the options and then set to something later, we need to get the actual value used otherwise we have the same issue.

Reply:
Null options seem to be OK for me - have to put `null` in the yaml file to get `None` in the options in Python, but I think that's a minor annoyance. I am finding that some values are being written as `numpy.core.multiarray.scalar` (which look odd and I don't know if they'll load correctly) where I'd think they should be `float` - I'll have a look at why that is...